tests/data/test_callbacks.py

```diff
@@ -30,6 +33,52 @@ def _rand_image():
     return Image.fromarray(np.random.randint(0, 255, (196, 196, 3), dtype="uint8"))
 
 
+def test_base_data_fetcher(tmpdir):
+
+    class CheckData(BaseDataFetcher):
+
+        def check(self):
+            assert self.batches["val"]["load_sample"] == [0, 1, 2, 3, 4]
+            assert self.batches["val"]["pre_tensor_transform"] == [0, 1, 2, 3, 4]
+            assert self.batches["val"]["to_tensor_transform"] == [0, 1, 2, 3, 4]
+            assert self.batches["val"]["post_tensor_transform"] == [0, 1, 2, 3, 4]
+            assert torch.equal(self.batches["val"]["collate"][0], tensor([0, 1, 2, 3, 4]))
+            assert torch.equal(self.batches["val"]["per_batch_transform"][0], tensor([0, 1, 2, 3, 4]))
+            assert self.batches["train"] == {}
+            assert self.batches["test"] == {}
+            assert self.batches["predict"] == {}
+
+    class CustomDataModule(DataModule):
+
+        @staticmethod
+        def configure_data_fetcher():
+            return CheckData()
+
+        @classmethod
+        def from_inputs(cls, train_data: Any, val_data: Any, test_data: Any, predict_data: Any) -> "CustomDataModule":
+
+            preprocess = cls.preprocess_cls()
+
+            return cls.from_load_data_inputs(
+                train_load_data_input=train_data,
+                val_load_data_input=val_data,
+                test_load_data_input=test_data,
+                predict_load_data_input=predict_data,
+                preprocess=preprocess,
+                batch_size=5
+            )
+
+    dm = CustomDataModule.from_inputs(range(5), range(5), range(5), range(5))
+    data_fetcher: CheckData = dm.data_fetcher
+
+    with data_fetcher.enable():
+        _ = next(iter(dm.val_dataloader()))
```

Comment:
I'd add check to make sure it's populated before the reset

Reply:
Will do so in another PR. Might contains some logic.